nixos/modules/services/networking/dnscrypt-proxy2.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }: with lib;
+
+let
+  cfg = config.services.dnscrypt-proxy2;
+
+  json = pkgs.writeText "dnscrypt-proxy.json" (builtins.toJSON cfg.config);
+
+  toml = pkgs.runCommand "dnscrypt-proxy.toml" {} ''
+    ${pkgs.remarshal}/bin/json2toml < ${json} > $out
+  '';
+in
+
+{
+  options.services.dnscrypt-proxy2 = {
+    enable = mkEnableOption "dnscrypt-proxy2";
+
+    config = mkOption {
+      description = ''
+        Attrset that is converted and passed as TOML config file.
+        For available params, see: https://git.io/fxBne
```

Comment:
Use the docbook link syntax `<link xlink:href="https://example.com"/>`. Also I don't think the url needs to be shortened, a slightly longer link isn't a problem, and lets the user know where they're going